pkg/messaging/direct_messaging.go

```diff
@@ -133,6 +136,33 @@ func (d *directMessaging) invokeRemote(ctx context.Context, targetID string, req
 	defer cancel()
 
 	ctx = diag.AppendToOutgoingGRPCContext(ctx, sc)
+
+	metadata := map[string][]string{}
+
+	var forwardedHeaderValue string
+
+	hostAddress, err := utils.GetHostAddress()
```

Comment:
host address and host name are not going to change so I'd make these properties on the struct and assign in `NewDirectMessaging(...)` rather than looking up on each invocation.

Reply:
That's better, added them to struct.